packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -184,16 +184,99 @@ export class EthereumAnchorService implements AnchorService {
       })
     )
 
+    return requestWithError.pipe(
+      expand(() => {
+        const now = new Date().getTime()
+        if (now > maxTime) {
+          throw new Error('Exceeded max anchor polling time limit')
+        } else {
+          return timer(this.pollInterval).pipe(concatMap(() => requestWithError))
+        }
+      }),
+      map((response) => this.parseResponse(cidStream, response))
+    )
+  }
+
+  newPollForAnchorResponse(task: () => Promise<any>): Observable<CASResponse> {
```

Comment:
is this used anywhere?  Is this actually meant to be merged?

Reply:
nope just for the demo